09_json/nickolee/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	puppy "github.com/anz-bank/go-course/09_json/nickolee/pkg/puppy"
+	store "github.com/anz-bank/go-course/09_json/nickolee/pkg/puppy/store"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var out io.Writer = os.Stdout
+
+var (
+	args     = os.Args[1:]
+	fileName = kingpin.Flag("data", "This flag supplies a puppy JSON file name").Short('d').String()
+)
+
+func main() {
+	// parse flags and read json from file
+	fileData := parseFlagsAndLoadFile() // fileData var to store raw bytes read in from file
+
+	// parse/unmarshal the json and we now have []puppy (Go objects)
+	puppies := unmarshalPuppies(fileData)
+
+	// instantiate new puppy store
+	store := store.NewMapStore()
+
+	// create some puppies loaded in from json file and store in store (no pun intended lol)
+	for _, pup := range puppies {
+		pup := pup // to avoid scopelint error: Using a reference for the variable on range scope `pup`
+		createdPuppy, _ := store.CreatePuppy(&pup)
+		fmt.Fprintf(out, "Puppy with ID %d has been created\n", createdPuppy)
+		retrievedPuppy, _ := store.ReadPuppy(createdPuppy)
+		fmt.Fprintln(out, "Retrieved puppy:", retrievedPuppy)
+	}
+}
+
+// parse flags and load file
+func parseFlagsAndLoadFile() (fileData []byte) {
```

Comment:
Functions that can error out should return an error, not panic.

Reply:
Could not figure out tests and test coverage and how to do this so for now have stuck with the panics. Might have a chat with you further about this.